###### Pandas Assignment 1: Setlements and Demography of Jigawa State Nigeria

by Engr Chid,  28 June 2020

This document contain analysis of Jigawa State Demography.

In 2006, the Nigeria Government embarked on a national Population Census accross the 36 States of Nigeria. This exercise will help the government to know how to distribute social amenities, campaign for elections and ration scarce resources of the Country.
Gigawa State happens to be amoung the states in the north that have high population. They have over 5,000 wardd in over 40 LGAs.
The Langitude and Lattitude is also of interest as it varies so much that one can go off Jigawa or be in Jigawa without being sure, if one only relies on Geolocation readings.

Given this kind of demographic variations, we want to know the Following:
    -What is the total Population of Jigawa State (nigeria)
    -What total number of Males and Females in the State
    -Which Ward has the Highest and the Lowest Populated wards
    -What population distribution is living at the remotest part of the State


# The data
The data consists of total population of the different Wards in Jigawa State. The data Population is split into male and female per ward. 

The data was taken in August 2006 by the national Population Commission. The data was actually in CSV formart but was converted to XLS format for easier reading.

In [1]:
from pandas import *
data = read_excel('jigawastategatessettlement.xls')
data

FID State Code   State  Lga Code          Lga Ward Code      Ward  \
0     19398         JI  Jigawa     18012        Dutse    JI0602     Chamo   
1     19399         JI  Jigawa     18002        Taura    JI2610     Taura   
2     19400         JI  Jigawa     18007  Kiri Kasama    JI1801  Baturiya   
3     19401         JI  Jigawa     18025       Kiyawa    JI1911    Tsirma   
4     19402         JI  Jigawa     18021     Biriniwa    JI0404   Diginsa   
...     ...        ...     ...       ...          ...       ...       ...   
8251  27649         JI  Jigawa     18022  Birnin Kudu    JI0310     Wurno   
8252  27650         JI  Jigawa     18025       Kiyawa    JI1906    Katuka   
8253  27651         JI  Jigawa     18025       Kiyawa    JI1907    Kiyawa   
8254  27652         JI  Jigawa     18025       Kiyawa    JI1907    Kiyawa   
8255  27653         JI  Jigawa     18006        Gumel    JI0911     Zango   

            Settlement      Settleme_1 Settleme_2   Latitude  Longitude  \
0     BUA_JI0602_10400         Abaderi        BUA  11.966424   9.303389   
1     SSA_JI2610_96056         Abakura        SSA  12.210321   9.252232   
2     HA_JI1801_366586        Abangawa         HA  12.571800  10.299900   
3     SSA_JI1911_71667         Abarawa        SSA  11.923330   9.605391   
4     SSA_JI0404_96058           Abari        SSA  12.772188   9.940747   
...                ...             ...        ...        ...        ...   
8251  HA_JI0310_472686         Zaramba         HA  11.389700   9.560000   
8252  HA_JI1906_434302  Zawayen Fulani         HA  11.804670   9.589963   
8253  HA_JI1907_434322  Zawayen Fulani         HA  11.800895   9.590530   
8254  SSA_JI1907_72554  Zawayen Kanawa        SSA  11.797315   9.583456   
8255  BUA_JI0911_10670            Zuwo        BUA  12.628383   9.393230   

      Female  Male  Total  
0        911   869   1780  
1        188   185    373  
2        179   166    345  
3        151   147    298  
4        332   316    648  
...      ...   ...    ...  
8251     880   855   1735  
8252     169   166    335  
8253     110   108    218  
8254     150   147    297  
8255   10130  8715  18845  

[8256 rows x 15 columns]


## The range of the problem
The column of interest is the Column Total.

In [2]:
TotalColumn = data['Total']

In [ ]:
Now we cant get the Total population of Jigawa State:

In [3]:
TotalColumn.sum()

5718275

In [ ]:
Next is to get the Total Population of Males in that State.

In [18]:
TotalMale=data['Male']
TotalMale.sum()

2799490

In [11]:
TotalFemale=data['Female']
TotalFemale.sum()

2918785

In [ ]:
So what percentage of the Population are males?

### Lets use the mean and Median to get the average population of the State.

In [12]:
TotalColumn.mean()


249.0

In [13]:
TotalColumn.median()

249.0

In [ ]:
Since the mean and median are very close, it means that the population is evenly distributed.


## The most populated Ward
To get this valuue, we sort the Population Total column in Ascending order.

In [14]:
data.sort_values('Total')

FID State Code   State  Lga Code            Lga Ward Code  \
3066  22464         JI  Jigawa     18023      Maigatari    JI2004   
6468  25866         JI  Jigawa     18030  Sule Tankakar    JI2503   
6474  25872         JI  Jigawa     18021       Biriniwa    JI0410   
2107  21505         JI  Jigawa     18026          Jahun    JI1403   
6475  25873         JI  Jigawa     18030  Sule Tankakar    JI2508   
...     ...        ...     ...       ...            ...       ...   
7795  27193         JI  Jigawa     18004         Babura    JI0201   
7837  27235         JI  Jigawa     18025         Kiyawa    JI1907   
7935  27333         JI  Jigawa     18006          Gumel    JI0907   
7854  27252         JI  Jigawa     18012          Dutse    JI0606   
8209  27607         JI  Jigawa     18022    Birnin Kudu    JI0301   

                Ward        Settlement         Settleme_1 Settleme_2  \
3066          Galadi  HA_JI2004_346761              HA_21         HA   
6468       Dangwanki  HA_JI2503_494384     Rugagen Kareka         HA   
6474          Matamu  HA_JI0410_497805  Rugan Lamido Audu         HA   
2107           Gauza  HA_JI1403_473210        Gidan Yusai         HA   
6475  Sule Tankarkar  HA_JI2508_348318     Rugan Mai Kudi         HA   
...              ...               ...                ...        ...   
7795          Babura   BUA_JI0201_9981             Babura        BUA   
7837          Kiyawa  BUA_JI1907_51763            Darkawa        BUA   
7935           Gusau  BUA_JI0907_60985              Gusau        BUA   
7854           Kachi  BUA_JI0606_10433             Fagoji        BUA   
8209     Birnin Kudu  BUA_JI0301_33560           Tsangaya        BUA   

       Latitude  Longitude  Female   Male  Total  
3066  12.803499   9.565100       0      0      0  
6468  12.826414   9.220886       0      0      0  
6474  12.950692  10.238678       0      0      0  
2107  12.067753   9.749636       0      0      0  
6475  12.651229   9.263212       0      0      0  
...         ...        ...     ...    ...    ...  
7795  12.772259   9.014382   15896  15788  31684  
7837  11.785581   9.605448   16937  16694  33631  
7935  12.618898   9.387879   18558  16011  34569  
7854  11.717812   9.361268   21251  19413  40664  
8209  11.451764   9.476027   21315  20919  42234  

[8256 rows x 15 columns]

#The result indicates that those wards at Low level lands (smaller Longitude and latitude) and more densly populated 
than those at the highlands and mountain tops (with Higher Latitudes and Longitudes)

In [ ]:
Rate of Males in each Popultion.
We have seen that there are more males than females in Jigawa State. We want to get the rate of males in each Ward.

In [16]:
MaleColumn = data['Male']
Totalcolum = data['Total']
data['MaleRate'] = MaleColumn * 100 / Totalcolum
data

FID State Code   State  Lga Code          Lga Ward Code      Ward  \
0     19398         JI  Jigawa     18012        Dutse    JI0602     Chamo   
1     19399         JI  Jigawa     18002        Taura    JI2610     Taura   
2     19400         JI  Jigawa     18007  Kiri Kasama    JI1801  Baturiya   
3     19401         JI  Jigawa     18025       Kiyawa    JI1911    Tsirma   
4     19402         JI  Jigawa     18021     Biriniwa    JI0404   Diginsa   
...     ...        ...     ...       ...          ...       ...       ...   
8251  27649         JI  Jigawa     18022  Birnin Kudu    JI0310     Wurno   
8252  27650         JI  Jigawa     18025       Kiyawa    JI1906    Katuka   
8253  27651         JI  Jigawa     18025       Kiyawa    JI1907    Kiyawa   
8254  27652         JI  Jigawa     18025       Kiyawa    JI1907    Kiyawa   
8255  27653         JI  Jigawa     18006        Gumel    JI0911     Zango   

            Settlement      Settleme_1 Settleme_2   Latitude  Longitude  \
0     BUA_JI0602_10400         Abaderi        BUA  11.966424   9.303389   
1     SSA_JI2610_96056         Abakura        SSA  12.210321   9.252232   
2     HA_JI1801_366586        Abangawa         HA  12.571800  10.299900   
3     SSA_JI1911_71667         Abarawa        SSA  11.923330   9.605391   
4     SSA_JI0404_96058           Abari        SSA  12.772188   9.940747   
...                ...             ...        ...        ...        ...   
8251  HA_JI0310_472686         Zaramba         HA  11.389700   9.560000   
8252  HA_JI1906_434302  Zawayen Fulani         HA  11.804670   9.589963   
8253  HA_JI1907_434322  Zawayen Fulani         HA  11.800895   9.590530   
8254  SSA_JI1907_72554  Zawayen Kanawa        SSA  11.797315   9.583456   
8255  BUA_JI0911_10670            Zuwo        BUA  12.628383   9.393230   

      Female  Male  Total   MaleRate  
0        911   869   1780  48.820225  
1        188   185    373  49.597855  
2        179   166    345  48.115942  
3        151   147    298  49.328859  
4        332   316    648  48.765432  
...      ...   ...    ...        ...  
8251     880   855   1735  49.279539  
8252     169   166    335  49.552239  
8253     110   108    218  49.541284  
8254     150   147    297  49.494949  
8255   10130  8715  18845  46.245689  

[8256 rows x 16 columns]

In [ ]:
#also the percentage of Males in that state is also important.

In [17]:
perc = TotalMale.sum()*100/TotalColumn.sum()
print(perc)

48.9568969663054


In [ ]:
This lets us kniow that about 49 percent of the population are male.


# Conclusion
Jigawa State is an interesting State with diversifies demography. Given the Census of 2006, we have seen that The total Population of Jigawa State is about 5.7 million (5,718,275). There are more Females in Jigawa State than there are Males, with the percentage of 51/49.
Another interesting discovery is that the Wards located at Low lands (smaller Longitude and latitude) and more densly populated than those at the highlands and mountain tops (with Higher Latitudes and Longitudes). 